### PR TITLE
[JIT] Update docs for recently added features

### PR DESCRIPTION
**Pull request by SplitInfinity**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #45234 [JIT] Add recursive scripting for class type module attributes
* #45262 [JIT] Add API for ignoring arbitrary module attributes
* #45261 [JIT] Enable @unused syntax for ignoring properties
* #45233 [JIT] Add test for ignored class type property
* **#45232 [JIT] Update docs for recently added features**

**Summary**
This commit updates the TorchScript language reference to include
documentation on recently-added TorchScript enums. It also removed
`torch.no_grad` from the list of known unsupported `torch` modules and
classes because it is now supported.

**Test Plan**
Continuous integration.

Differential Revision: [D23971884](https://our.internmc.facebook.com/intern/diff/D23971884)

### CODE REVIEW

**Comment by SplitInfinity** [comment]
Where's a good place to mention `property` support? They're not part of the core Python language so `jit_python_reference.rst` is not the right place to put documentation about them. A new section in `jit_language_reference.rst`, maybe?

File: docs/source/jit_unsupported.rst [hunk 1]
```diff
@@ -87,6 +87,5 @@ we suggest using :meth:`torch.jit.trace`.
   * :class:`torch.nn.RNN`
   * :class:`torch.nn.AdaptiveLogSoftmaxWithLoss`
   * :class:`torch.autograd.Function`
-  * :class:`torch.autograd.no_grad`
```
Comment (SplitInfinity) [inline docs/source/jit_unsupported.rst:90]: Is there a good place to mention that `record_function` is supported now?

File: docs/source/jit_language_reference.rst [hunk 2]
```diff
@@ -346,6 +348,73 @@ like any other TorchScript type:
...
+    p = Color.RED
+    q = Color.GREEN
+    print(enum_comp(p, q))
+
+The type of the values of an enum must be ``int``, ``float``, or ``str``:
+
+::
+    from enum import Enum
+
+    global IntEnum
```
[the remark this answers is not in the document]
Comment (SplitInfinity) [reply docs/source/jit_language_reference.rst]: Fixed.

File: docs/source/jit_language_reference.rst [hunk 3]
```diff
@@ -346,6 +348,73 @@ like any other TorchScript type:
...
+        GREEN = 2
+
+    @torch.jit.script
+    def enum_comp(x: Color, y: Color) -> bool:
+        return x == y
+
+After an enum is defined, it can be used in both TorchScript and Python interchangeably
+like any other TorchScript type:
+
+::
```
[the remark this answers is not in the document]
Comment (SplitInfinity) [reply docs/source/jit_language_reference.rst:380]: Fixed.

File: docs/source/jit_language_reference.rst [hunk 4]
```diff
@@ -346,6 +348,74 @@ like any other TorchScript type:
...
+.. _torchscript-enums:
+
+TorchScript Enums
+^^^^^^^^^^^^^^^^^^^
+
+.. warning::
+
+    TorchScript enum support is experimental..
+
+Python enums can be used in TorchScript without any extra annotation or code:
```
Comment (eellison) [inline docs/source/jit_language_reference.rst:362]: Personally I would cut down on this section a bit - it's pretty long for what is an easy & simple feature (enums).   see how short the named tuple section is

**Review by eellison — COMMENTED** [review]
Looks good, a couple comments

File: docs/source/jit_language_reference.rst [hunk 5]
```diff
@@ -346,6 +348,74 @@ like any other TorchScript type:
...
+.. _TorchScript Enums:
+.. _TorchScript Enums:
+.. _torchscript-enums:
+
+TorchScript Enums
+^^^^^^^^^^^^^^^^^^^
+
+.. warning::
+
+    TorchScript enum support is experimental..
```
[the remark this answers is not in the document]
Comment (gmagogsfm) [reply docs/source/jit_language_reference.rst:360]: lol

File: docs/source/jit_language_reference.rst [hunk 6]
```diff
@@ -346,6 +348,74 @@ like any other TorchScript type:
     print(sum_pair(p))
 
 
+.. _TorchScript Enums:
+.. _TorchScript Enums:
```
[the remark this answers is not in the document]
Comment (SplitInfinity) [reply docs/source/jit_language_reference.rst:352]: I copied over the class types section, I don't know how to write `.rst` lol

File: docs/source/jit_language_reference.rst (hunk 4 above)
Comment (SplitInfinity) [reply docs/source/jit_language_reference.rst:362]: What do you think I should cut out? Most of it is code samples.

File: docs/source/jit_language_reference.rst [hunk 7]
```diff
@@ -346,6 +348,74 @@ like any other TorchScript type:
...
+    class Color(Enum):
+        RED = 1
+        GREEN = 2
+
+    @torch.jit.script
+    def return_enum(cond: bool):
+        if cond:
+            return Color.RED
+        else:
+            return Color.GREEN
```
Comment (eellison) [inline docs/source/jit_language_reference.rst:393]: And then maybe include this example where you access the field name `Color.RED` or make it part of the other function

**Review by eellison — APPROVED** [review]
Meant to accept - these are just suggestions though, it looks good obviously feel free to keep whatever

**Pull request closed without merging**